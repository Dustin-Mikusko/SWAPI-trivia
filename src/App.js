import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      user: {
        name: '',
        quote: '',
        rank: '',
        favoriteCharacters: []
      },
      movies: []
      // {
      //   title: '',
      //   episode: null,
      //   releaseDate: null,
      //   characters: [],
      //   image: '../images/movie.jpeg',
      //   openingCredits: ''
      // }
    }
  }

  componentDidMount() {
    fetch('https://swapi.co/api/films/')
      .then(response => response.json())
      .then(data => {
        let moviesAndCharactersPromises = data.results.map(movie => {

          return movie.characters.map(character => {
            let world = this.fetchWorld(character);
            console.log('world', world);
            return fetch(character)
            .then(response => response.json())
            .then(character => ({
              name: character.name,
              world: this.world.name,
              population: this.world.population,
              species: this.fetchSpecies(character),
              relatedFilms: this.fetchFilms(character)
            }))
            .catch(error => console.log(error))
          }
          )
          // return {
          //   title: movie.title,
          //   episode: movie.episode_id,
          //   releaseDate: release_date,
          //
          // }
        })
        return Promise.all(moviesAndCharactersPromises);
        // .catch(error => console.log(error)
      // })
    })
    .then(data => console.log(data))
  }

    fetchWorld(character) {
      let world =
      fetch(character.homeworld)
        .then(response => response.json())
        .then(data => ({
          name: data.name,
          population: data.population
        }))
        .catch(error => console.log(error))
        console.log('world', world);
      return world;
    }

    fetchSpecies(character) {
      let species =
      fetch(character.species)
        .then(response => response.json())
        .then(data => data.name)
        .catch(error => console.log(error))
      return species;
    }

    fetchFilms(character) {
      return character.films.map(film => {
        return fetch(film)
          .then(response => response.json())
          .then(data => data.title)
          .catch(error => console.log(error))
      })
    }

  render() {
    return (
      <div className="App">
      <header className="App-header">
      <img src={logo} className="App-logo" alt="logo" />
      <p>
      Edit <code>src/App.js</code> and save to reload.
      </p>
      <a
      className="App-link"
      href="https://reactjs.org"
      target="_blank"
      rel="noopener noreferrer"
      >
      Learn React
      </a>
      </header>
      </div>
    )
  }
}

export default App;
