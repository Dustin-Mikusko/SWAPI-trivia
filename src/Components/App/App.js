import React, { Component } from 'react';
import './App.css';
import Form from '../Form/Form';
import { Route, Redirect } from 'react-router-dom';
import MovieContainer from '../MovieContainer/MovieContainer'


class App extends Component {
  constructor() {
    super();
    this.state = {
      user: {
        name: '',
        quote: '',
        rank: '',
        favoriteCharacters: [],
        loggedIn: false,
      },
      movies: []
    }
  }

  addUser = newUser => {
    console.log(newUser)
    this.setState({ user: newUser })

  }

  componentDidMount() {
    fetch('https://swapi.co/api/films/')
      .then(response => response.json())
      .then(data => {
        return data.results.map(movie => {
          return {
            title: movie.title,
            episode: movie.episode_id,
            releaseDate: this.parseReleaseDate(movie.release_date),
            image: '../images/movie.jpeg',
            openingCredits: movie.opening_crawl
          }
       })
      })
      .then(movies => this.setState({ movies }))
      .catch(err => this.setState(err))
  }

  parseReleaseDate = date => {
    return date.split('-')[0];
  }

    fetchWorld = (character) => {
      let world =
      fetch(character.homeworld)
        .then(response => response.json())
        .then(data => ({
          name: data.name,
          population: data.population
        }))
        .catch(error => console.log(error))
        // console.log('world', world);
      return world;
    }

    fetchSpecies = (character) => {
      let species =
      fetch(character.species)
        .then(response => response.json())
        .then(data => data.name)
        .catch(error => console.log(error))
      return species;
    }

    fetchFilms = (character) => {
      return character.films.map(film => {
        return fetch(film)
          .then(response => response.json())
          .then(data => data.title)
          .catch(error => console.log(error))
      })
    }

    render() {
      if (this.state.user.loggedIn) {
        console.log(this.state.movies)
        return (
          <main>
            <Redirect to="/movies" />
            <Route exact path='/movies' render={() => <MovieContainer movies={this.state.movies} user={this.state.user}/> } />
          </main>
        )
      } else {
          return (
            <main>
              <Route exact path='/' render={ () => <Form addUser={this.addUser} /> } />
            </main>
         )
      }
    }
  }

    // if (this.state.user.loggedIn) {
    //   return (
    //     <main>
    //       <Redirect to='/movies' />
    //       <Route exact path='/movies' render={() => <MovieContainer movies={this.state.movies}/> } />
    //     </main>
    //   )
    // } else {
    //   return (<Form addUser={this.addUser} userLogin={this.userLogin} />)
    // }




    // return (
    //   <body>
    //     <Route exact path='/' render={() => this.state.user.loggedIn ? <Redirect to='/movies' /> : <Form addUser={this.addUser} userLogin={this.userLogin}/> } />
    //   </body>
    //
    // )



export default App;






// return movie.characters.map(character => {
//   return fetch(character)
//   .then(response => response.json())
//   .then(character => {
//     let world = this.fetchWorld(character);
//     return {
//       name: character.name,
//       world: world.name,
//       population: world.population,
//       species: this.fetchSpecies(character),
//       relatedFilms: this.fetchFilms(character)
//     }
//   })
//   .catch(error => console.log(error))
// })

// return Promise.all(moviesAndCharactersPromises)
//           .then(data => console.log('line 59:', data))
//           .catch(err => console.log(err))
