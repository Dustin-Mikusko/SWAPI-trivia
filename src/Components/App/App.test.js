import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import ReactDOM from 'react-dom';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should create main App component', () => {
    // const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should update state of user when addUser is called', () => {
    // const wrapper = shallow(<App />);
    const mockState = {
      name: '',
      quote: '',
      rank: '',
      favoriteCharacters: []
    }
    const newUser = {
      name: 'John Adams',
      quote: 'Do or do not.  There is no try.',
      rank: 'intermediate',
      favoriteCharacters: []
    }
    wrapper.setState({user: mockState});
    expect(wrapper.state('user')).toEqual(mockState);
    wrapper.instance().addUser(newUser)
  })

  // it('should call componentDidMount after rendering', () => {
  //   const mockComponentDidMount = jest.fn();
  //   expect(wrapper.instance().mockComponentDidMount).toHaveBeenCalled();
  // })

  it('should parse a date to return the year', () => {
    const date = '2000-1-2';
    expect(wrapper.instance().parseReleaseDate(date)).toEqual('2000')
  })

  // it('should fetch data for homeworld', () => {
  //   const fetchWorld = jest.fn();
  //   expect(wrapper.instance().fetchWorld).toHaveBeenCalled();
  // })

  // it('should fetch data for character species', () => {
  //   const fetchSpecies = jest.fn();
  //   expect(wrapper.instance().fetchSpecies).toHaveBeenCalled();
  // })

  // it('should fetch data for character's films', () => {
  //   const fetchFilms = jest.fn();
  //   expect(wrapper.instance().fetchFilms).toHaveBeenCalled();
  // })

  


})
