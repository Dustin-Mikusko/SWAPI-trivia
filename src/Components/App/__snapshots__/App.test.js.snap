// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should create main App component 1`] = `
<Form
  addUser={[Function]}
  user={
    Object {
      "favoriteCharacters": Array [],
      "name": "",
      "quote": "",
      "rank": "",
    }
  }
/>
`;
